### ISSUE TITLE
Add AWS Organizations ID to CloudWatch Events Permission

### ISSUE THREAD

**Issue by connor-tyndall** [terraform-providers/terraform-provider-aws#6254]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Due to recent release from AWS, [Amazon CloudWatch Events Adds the Ability to Share Events Across All Accounts in an Organization](https://aws.amazon.com/about-aws/whats-new/2018/10/amazon-cloudwatch-events-adds-the-ability-to-share-events-across-all-accounts-in-an-organization/), the Terraform resource `aws_cloudwatch_event_permission` needs to support AWS organizations ID or a new resource, such as `aws_cloudwatch_event_policy` needs to be created similar to [CloudFormation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-events-eventbuspolicy.html).

### New or Affected Resource(s)

* aws_cloudwatch_event_permission
* aws_cloudwatch_event_policy

### Potential Terraform Configuration

```hcl
resource "aws_cloudwatch_event_permission" "OrgAccountAccess" {
  principal    = "o-12abc3de4fg"
  statement_id = "OrgAccountAccess"
}
```

**OR**

```hcl
resource "aws_cloudwatch_event_policy" "OrgAccountAccess" {
  name = "OrgAccountAccess
  
  policy = <<EOF
{
  "Version": "2012-10-17",
  "Statement": [
     {
         "Action": [
            "events:PutEvents"
         ],
         "Effect": "Allow",
         "Resource": "*",
             "Condition": {
                  "StringEquals": {
                      "aws:PrincipalOrgID": "o-12abc3de4fg"
                   }
               }
          }
     ]
}
POLICY
}
```

### References

https://aws.amazon.com/about-aws/whats-new/2018/10/amazon-cloudwatch-events-adds-the-ability-to-share-events-across-all-accounts-in-an-organization/

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-events-eventbuspolicy.html

**Comment by bflad** [comment]
Pull request submitted: #6261

**Comment by bflad** [comment]
Support for a new `condition` configuration block in `aws_cloudwatch_event_permission` has been merged and will release with version 1.42.0 of the AWS provider, likely today or tomorrow. 👍

**Comment by bflad** [comment]
This has been released in [version 1.42.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1420-october-31-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
